run/runner_test.go

```diff
@@ -0,0 +1,55 @@
+package run
+
+import (
+	"github.com/stretchr/testify/assert"
+	"strings"
+	"testing"
+)
+
+func TestPruneDirsWithFilter(t *testing.T) {
+	runner := Runner{
+		RepoPath:        "/repo/",
+		RepoPathFilters: []string{"run", "webserver"},
+	}
+
+	dirs := strings.Split(`/repo/.git
+/repo/git
+/repo/kube
+/repo/log
+/repo/Makefile
+/repo/manifests
+/repo/metrics
+/repo/run
+/repo/static
+/repo/sysutil
+/repo/templates
+/repo/webserver
+`, "\n")
+
+	prunedDirs := runner.pruneDirs(dirs)
+	assert.Len(t, prunedDirs, 2)
+}
+
+func TestPruneDirsWithoutFilter(t *testing.T) {
+	runner := Runner{
+		RepoPath:        "/repo/",
+		RepoPathFilters: []string{},
+	}
+
+	dirs := strings.Split(`/repo/.git
+/repo/git
+/repo/kube
+/repo/log
+/repo/Makefile
+/repo/manifests
+/repo/metrics
+/repo/run
+/repo/static
+/repo/sysutil
+/repo/templates
+/repo/webserver
+`, "\n")
+
+	prunedDirs := runner.pruneDirs(dirs)
+	assert.Len(t, prunedDirs, 13)
```

Comment:
I actually bothered to count and think about the empty line as an entry to make 13. That could be `len(dirs)` and dirs could go outside of the functions but that is just personal style I guess.